### ISSUE TITLE
Features

### ISSUE THREAD

**Issue by alg** [alg/hack_night#2]
Вот несколько фич и изменений, которые пришли в голову вчера:
- вместо project message board показывать активный статус участников (Статус можно ставить через твиттер используя хэштег приложения -- например, #hn. Статус валиден Х минут.)
- Suggest new project открывает инлайн форму с названием проекта и его хт
- В деталях проекта показывать статусы участников
- В деталях проекта показывать join и leave
- Когда меняется статус из приложения, слать его в твиттер
- Рейтинги для проектов (Рейтинг = сумме количества участников в каждом хн)
- Рейтинг пользователей (= количество хн участия. Может стимулировать к нажатию на attend)
- Максимум участников (Организатор может указать сколько участников он может принять. Когда достигнут лимит, кнопка attend гасится и подписыватся, что мест нет, но могут освободиться. Люди, которые сказали, что придут, но не пришли както отмечаются. При достижении кагого-то лимита, человек считается ненадежным. Статус ненадежного снимается после нескольких успешных хн.)
- Не показывать suggest project, join, leave, attend и disable message board когда не залогинен

Мысли в слух...

**Comment by iosadchiy** [comment]
То есть вместо борды для общей ленты используем твиттер со всеми вытекающими. А что делать с сообщениями, привязанными к проектам?

Ограничение числа участников тянет за собой необходимость подсчета, сколько раз "обещал, но не пришел", а заодно и рейтинги. Плюс необходимо кого-то нагружать, чтобы отмечал пришедших.

Чтобы не делить страницы на "для пользователей" и "для анонимусов", может, отправлять сразу на твиттер за авторизацией? Я тут быстренько попробовал - получается интересно, только немного неожиданно: заходил на hacknight, попал на твиттер.

**Comment by alg** [comment]
> То есть вместо борды для общей ленты используем твиттер со всеми вытекающими. А что делать с сообщениями, привязанными к проектам?

Нет, общий борд остается, а вот вместо per-project мы используем просто статусы, говорящие кто и чем занимается. Этого должно быть вполне достаточно.

> Ограничение числа участников тянет за собой необходимость подсчета, сколько раз "обещал, но не пришел", а заодно и рейтинги. Плюс необходимо кого-то нагружать, чтобы отмечал пришедших.

Верно. Единственное, что смущает -- это отметка не пришедших. Давай не будем ее делать. В остальном мне кажется идея здравая. Юра при всем желании, к примеру, не смог бы разместить 20 человек. А если бы и смог, то пользы от этого было бы значительно меньше, чем если их было бы 12, но все с рабочим местом.

> Чтобы не делить страницы на "для пользователей" и "для анонимусов", может, отправлять сразу на твиттер за авторизацией? Я тут быстренько попробовал - получается интересно, только немного неожиданно: заходил на hacknight, попал на твиттер.

Мне кажется для анонимусов все же надо хоть и минимально. Во-первых, это удобно для людей, которые хотят понять стоит ли ввязываться прежде чем отдать свою инфу. Во-вторых, такой подход всем понятен и полностью прозрачен. Даже если только главная страница будет видна всем -- этого будет достаточно, чтобы дать правильное ощущение знакомости и понятности.
- А

**Comment by cutalion** [comment]
Я надеюсь, что не буду выглядеть белой вороной, но я не пользуюсь твиттером. Мне не понятна его философия и для меня было бы усложнением юзабилити, если придется регистрироваться через него, а потом еше и какие-то непонятные статусы отслеживать )

Как на счет сделать выбор при регистрации? Для меня идеальный вариант - оставить мыло и получать различные уведомления(а самый идеальный, если из почты и ответить можно)

Второй вопрос. В связи с вышесказанным. Зачем слать статус в твиттер? Что это дает?

Что нам дадут рейтинги по проектам? На сколько я понимаю, рейтинги пользователей могу стимулировать активность. А что на счет рейтингов по проектам? Мне кажется, что это лишнее на данном этапе. Да и при обсуждении в четверг мы похожую мысль отклонили.

**Comment by alg** [comment]
> Я надеюсь, что не буду выглядеть белой вороной, но я не пользуюсь твиттером. Мне не понятна его философия и для меня было бы усложнением юзабилити,

Если философия твиттера понятна дошкольницам с мобилками, то причина в чем-то другом.

> если придется регистрироваться через него, а потом еше и какие-то непонятные статусы отслеживать )

Тебе отслеживать и не надо. А я бы с удовольствием получал бы в твиттер ваши статусы и знал бы кто что делает. Вместо того, чтобы лезть и рефрешить страницу каждые 5 минут. Вот тебе и философия.

> Как на счет сделать выбор при регистрации?

В чем разница между зарегистрироваться в твиттере или в приложении?

> Для меня идеальный вариант - оставить мыло и получать различные уведомления(а самый идеальный, если из почты и ответить можно)

Если тебе хочется регистрироваться

> Второй вопрос. В связи с вышесказанным. Зачем слать статус в твиттер? Что это дает?

Все заинтересованные будут знать что ты делаешь. И для этого не надо будет идти в приложение и смотреть специально.

> Что нам дадут рейтинги по проектам? На сколько я понимаю, рейтинги пользователей могу стимулировать активность. А что на счет рейтингов по проектам? Мне кажется, что это лишнее на данном этапе. Да и при обсуждении в четверг мы похожую мысль отклонили.

Никакие из рейтингов сейчас не "на столе". Посмотри, пожалуйста, в пивотал.

**Comment by cutalion** [comment]
> Если философия твиттера понятна дошкольницам с мобилками, то причина в чем-то другом.

Неожиданно отличный аргумент.

> Тебе отслеживать и не надо. А я бы с удовольствием получал бы в твиттер ваши статусы и знал бы кто что делает. Вместо того, чтобы лезть и рефрешить страницу каждые 5 минут. Вот тебе и философия.

Слабый аргумент, никто и не собирается рефрешить каждые 5 минут.

> В чем разница между зарегистрироваться в твиттере или в приложении?

Разницы тут нет. Через твиттер удобнее - не надо ничего вводить. Но я бы хотел получать уведомления в почту, и даже готов ввести свое мыло ради этого.

Честно говоря, для меня будет неудобно пользоваться сервисом, если он будет завязан только на твиттер. Почему нельзя добавить логин через гугл?

**Comment by alg** [comment]
> > Если философия твиттера понятна дошкольницам с мобилками, то причина в чем-то другом.
> 
> Неожиданно отличный аргумент.

Хах. Не в обиду, конечно. Ты не пишешь СМС всем друзьям по одному одно и то же, а лишь один раз, но все заинтересованные получают. Вот и вся философия. :)

В нашем случае, есть люди, которые хотели бы знать чем ты занят сейчас (пусть только в рамках проекта).

> > Тебе отслеживать и не надо. А я бы с удовольствием получал бы в твиттер ваши статусы и знал бы кто что делает. Вместо того, чтобы лезть и рефрешить страницу каждые 5 минут. Вот тебе и философия.
> 
> Слабый аргумент, никто и не собирается рефрешить каждые 5 минут.

Для кого-то слабый, а кто-то вполне реально интересуется кто чем занимается.

> > В чем разница между зарегистрироваться в твиттере или в приложении?
> 
> Разницы тут нет. Через твиттер удобнее - не надо ничего вводить. Но я бы хотел получать уведомления в почту, и даже готов ввести свое мыло ради этого.
> 
> Честно говоря, для меня будет неудобно пользоваться сервисом, если он будет завязан только на твиттер. Почему нельзя добавить логин через гугл?

Потому что это дополнительные трудности. Один раз зашел через твиттер, второй -- через гугл и попал в разные accounts, которые теперь надо связывать.

Я не против разных логинов в принципе, я против них как что-то на что надо тратить время _сейчас_.

**Comment by cutalion** [comment]
,    .    -  
.
     . Stackoverflow   10
   .       -  
   . ..  ,   .

**Comment by iosadchiy** [comment]
2Саня: емейл добавить не так сложно (просто спросить после авторизации в
твиттере), можем записать эту фичу, реализовать по мере возможности. И будут
тебе оповещения на мыло.

2Алексей: основная сложность при связывании - это слияние данных двух
пользователей. Для простоты можно одного просто игнорировать, тогда эта фича
перейдет в разряд простых. Опять-таки, можем положить в nice to have.
